LambdaSchool/auth-i
# Amy Shackles

- [x] Fork / Clone repo
- [x] Download dependencies
- [x] Use mongoose middleware to implement password hashing
- [x] Use local express middleware and mongoose methods
- [x] Use Postman to test the API 
- [x] Post method at /api/register endpoint: 
       - Create a user using information from body of request, hash password before saving to DB
- [x] Post method at /api/login endpoint:
       - Use credentials sent inside body to log in user - on successful log in, create a new session for the user and send back a 'Logged in' message and a cookie that contains the user ID. If login fails, respond with the correct status code and the message: 'You shall not pass!'
- [x] Get method at /api/users endpoint:
      - If user is logged in, respond with an array of all the users contained in the database.  If the user is not logged in, respond with the correct status code and the message: 'You shall not pass!'  Use this endpoint to verify that the password is hashed before it is saved.

Stretch Problem:
- [x] Write a piece of global middleware that ensures a user is logged in when accessing any route prefixed by /api/restricted.  For instance, /api/restricted/something, /api/restricted/other, and /api/restricted/a should all be protected by the middleware; only logged in users should be able to access these routes
- [x] Build a React application that implements components to register, login, and view a list of users
